src/client_side_request.cc

```diff
@@ -530,6 +536,7 @@ ClientRequestContext::hostHeaderVerify()
         debugs(85, 3, "FAIL on validate URL domain length " << http->request->url.host() << " matches Host: " << host);
```

Comment:
AFAICT, the strlen() check above this line used to work for bracketed IPv6 addresses because url.host() returns a bracketed variation and `host` kept its brackets. With PR changes, will that strlen() check always fail for bracketed IPv6 addresses?